packages/firestore/test/integration/api/type.test.ts

```diff
@@ -47,6 +47,12 @@ apiDescribe('Firestore', (persistence: boolean) => {
     });
   });
 
+  it('can read and write number fields', () => {
+    return withTestDb(persistence, db => {
+      return expectRoundtrip(db, { a: 1, b: NaN, c: Infinity, d: -0.0 });
```

Comment:
This doesn't actually make any assertion about the format of the `d` value. Is it worth separately asserting that in the forward direction `-0.0` renders as a string `'-0'`?

Reply:
`customDeepEqual` actually does assert that d is -0 with my changes below. If I just make those changes, this test fails.